### PR TITLE
CAT-1954 IF-Then brick without ELSE

### PR DESCRIPTION
**Pull request by p4p4**
**this reopens PR #1766 after rebasing** as requested by @thmq 

New Brick:
- IF-then without ELSE

reuses a lot of the standard if-else brick via inheritance. uses the same Action as the existing brick

This PR includes a TestCase

reviewed & refactored

Tests:
- [jenkins package 'test' results](https://jenkins.catrob.at/job/CatroidSinglePackageEmulatorTest/2170/)
- this pull does not break any tests
-- the two failing tests already fail in the current develop branch (June 23rd): [results for 'develop' branch](https://jenkins.catrob.at/job/CatroidSinglePackageEmulatorTest/2170/)

### CODE REVIEW

**Comment by Bioxar** [comment]
ok to test

**Comment by Bioxar** [comment]
ok to test

**Comment by Bioxar** [comment]
Please check this issue: [checkstyle](https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidTriggerPullRequestTest/317/)

**Comment by p4p4** [comment]
@Bioxar this seems to be an inconsistency between Checkstyle and the Formatter

Checkstyle does not allow to import with an asterisk:
    Using the '.*' form of import should be avoided - org.catrobat.catroid.content.bricks.*.

But that's exacty what our code formatter suggests! (before there were import statements for each and every Brick subclass, which were reported as unformatted.) 

Maybe @robertpainsi  has an idea what should be used? I don't see how to resolve this problem because either code is reported as unformatted or a checktyle warning is raised, no matter what I do.

File: catroid/src/org/catrobat/catroid/io/StorageHandler.java [hunk 1]
```diff
@@ -57,103 +57,7 @@
...
-import org.catrobat.catroid.content.bricks.UserListBrick;
-import org.catrobat.catroid.content.bricks.UserScriptDefinitionBrick;
-import org.catrobat.catroid.content.bricks.UserScriptDefinitionBrickElement;
-import org.catrobat.catroid.content.bricks.UserVariableBrick;
-import org.catrobat.catroid.content.bricks.VibrationBrick;
-import org.catrobat.catroid.content.bricks.WaitBrick;
-import org.catrobat.catroid.content.bricks.WhenBrick;
-import org.catrobat.catroid.content.bricks.WhenNfcBrick;
-import org.catrobat.catroid.content.bricks.WhenStartedBrick;
+import org.catrobat.catroid.content.bricks.*;
```
Comment (p4p4) [inline catroid/src/org/catrobat/catroid/io/StorageHandler.java]: how to do the import?
formatter says 
    import org.catrobat.catroid.content.bricks.*;
checkstyle says
   don't do this

**Comment by thmq** [comment]
There are several reasons why you should not use the import on demand (i.e. *). Although there is no difference in runtime performance the first and foremost, in my opinion, is that you cannot guarantee a clear dependency structure. Which means that if you refer to a class A which might be present in two or more sub packages your code will simply not compile. I am aware that in our structure this might not be the case at the moment, however, I think that it is best practice to prevent future bugs by thinking ahead and not taking these kinds of risks.

I have also read reasons for using *, such as the fact that it apparently simplifies refactoring. Anyway, I'd suggest you use the full list rather than the import on demand, but if anyone else has a good reason why we should use it please feel free to add your input. @robertpainsi @fwinkelbauer

**Comment by p4p4** [comment]
Ok I understand your reasons. -> changed imports  back & rebased. 
Let's see if the FormatBot complains ;)

**Comment by robertpainsi** [comment]
> I'd suggest you use the full list rather than the import on demand

I will create a pull request to disable wildcard imports in Android Studio asap.

**Comment by thmq** [comment]
ok to test

**Comment by robertpainsi** [comment]
@p4p4 please rebase your branch.

**Comment by p4p4** [comment]
@robertpainsi  done

**Comment by wslany** [comment]
Close https://jira.catrob.at/browse/CAT-1954

**Pull request merged**